shims/spark300/src/main/scala/com/nvidia/spark/rapids/shims/spark300/Spark300Shims.scala

```diff
@@ -193,4 +195,11 @@ class Spark300Shims extends SparkShims {
   override def getRapidsShuffleManagerClass: String = {
     classOf[RapidsShuffleManager].getCanonicalName
   }
+
+  override def injectQueryStagePrepRule(
+      extensions: SparkSessionExtensions,
+      ruleBuilder: SparkSession => Rule[SparkPlan]): Unit = {
+    // not supported in 3.0.0 but it doesn't matter because AdaptiveSparkPlanExec in 3.0.0 will
+    // never allow us to replace an Exchange node, so they just stay on CPU
```

Comment:
Should we throw an exception here to be sure of that assumption?

Reply:
Well, this code will still be called when we run with 3.0.0 but we can't inject the rule, because that feature isn't available in 3.0.0.

When the plugin runs against 3.0.0 with AQE on, our optimizer rules will only get applied to the children of any exchange nodes.